components/net/bluetooth_thread.rs

```diff
@@ -0,0 +1,128 @@
+}
+
+impl BluetoothManager {
+    pub fn new (receiver: IpcReceiver<BluetoothThreadMsg>) -> BluetoothManager {
+        BluetoothManager {
+            receiver: receiver,
+        }
+    }
+
+    fn start(&mut self) {
+        loop {
+            match self.receiver.recv().unwrap() {
+                BluetoothThreadMsg::RequestDevice(sender) => {
+                    self.request_device(sender)
+                }
+                BluetoothThreadMsg::GetCharacteristic(sender) => {
+                    self.get_characteristic(sender)
+                }
+                BluetoothThreadMsg::GetPrimaryService(sender) => {
+                    self.get_primary_service(sender)
+                }
+                BluetoothThreadMsg::GetDescriptor(sender) => {
+                    self.get_descriptor(sender)
+                }
+                BluetoothThreadMsg::GATTServerConnect(sender) => {
+                    self.gatt_server_connect(sender)
+                }
+                BluetoothThreadMsg::ReadValue(sender) => {
+                    self.read_value(sender)
+                }
+                BluetoothThreadMsg::WriteValue(sender) => {
+                    self.write_value(sender)
+                }
+                BluetoothThreadMsg::Exit => {
+                    break
+                }
+            }
+        }
+    }
+
+    fn request_device(&self, sender: IpcSender<BluetoothMsg>) {
+        let message = BluetoothMsg::BluetoothDevice {
+            id: String::from("ca:fe:c0:ff:ee:00"),
+            name: String::from("BLE Device (Mock)"),
+            device_class: 010918,
+            vendor_id_source: String::from("bluetooth"),
+            vendor_id: 29,
+            product_id: 4608,
+            product_version: 5174,
+            appearance: 832,
+            tx_power: 40,
+            rssi: -65
+        };
+        sender.send(message).unwrap();
+    }
+
+    pub fn get_characteristic(&self, sender: IpcSender<BluetoothMsg>) {
+        let message = BluetoothMsg::BluetoothCharacteristic {
+            uuid: String::from("00002a19-0000-1000-8000-00805f9b34fb"),
+            broadcast: true,
+            read: true,
+            write_without_response: false,
+            write: false,
+            notify: false,
+            indicate: false,
+            authenticated_signed_writes: false,
+            reliable_write: false,
+            writable_auxiliaries: false
+        };
+        sender.send(message).unwrap();
+    }
+
+    pub fn get_primary_service(&self, sender: IpcSender<BluetoothMsg>) {
+        let message = BluetoothMsg::BluetoothService {
+            uuid: String::from("0000180f-0000-1000-8000-00805f9b34fb"),
+            is_primary: true
+        };
+        sender.send(message).unwrap();
+    }
+
+    pub fn get_descriptor(&self, sender: IpcSender<BluetoothMsg>) {
+        let message = BluetoothMsg::BluetoothDescriptor {
+            uuid: String::from("00002902-0000-1000-8000-00805f9b34fb")
```

Comment:
"00002902-0000-1000-8000-00805f9b34fb" appears often, consider storing it as a static string.